src/test/java/com/patreon/euphrates/ScrubbingInputStreamTest.java

```diff
@@ -0,0 +1,47 @@
+package com.patreon.euphrates;
+
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+import org.apache.commons.io.IOUtils;
+
+import java.io.ByteArrayInputStream;
+
+import static org.junit.Assert.*;
+
+public class ScrubbingInputStreamTest extends TestCase {
+
+  public ScrubbingInputStreamTest(String testName) {
+    super(testName);
+  }
+
+  public static Test suite() {
+    return new TestSuite(ScrubbingInputStreamTest.class);
+  }
+
+  public void testScrubbingBytes() throws Exception {
+    byte[] in = {123, 0x10, 123, 0, 0xa, 0xb, 0xc, 123, 122};
+    ByteArrayInputStream bin = new ByteArrayInputStream(in);
+    ScrubbingInputStream sis = new ScrubbingInputStream(bin);
+    byte[] output = IOUtils.toByteArray(sis);
+    assertArrayEquals(new byte[]{123, 123, 0xa, 123, 122}, output);
+  }
+
+  public void testScrubbingMultibytes() throws Exception {
+    byte[] in = {123, -17, -65, -65, 123, 123};
```

Comment:
Might be good to write some tests with actual unicode strings that we expect to scrub (rather than encoding them by hand into byte arrays).  It's just gonna be easier to read the test without a unicode reference handy.  Should be able to just write a literal string and encode it to UTF-8.

Reply:
Good idea!